src/renderers/WebGLRenderer.js

```diff
@@ -1108,6 +1108,14 @@ function WebGLRenderer( parameters ) {
 		_currentMaterialId = - 1;
 		_currentCamera = null;
 
+		// onUpdate
+
+		scene.traverse( function( node ) {
+
+			node.onUpdate( _this, scene, camera );
```

Comment:
There are tradeoffs in either case. Go ahead and merge or modify to your liking. I have a PR for the light helpers in the queue.

If you are having second thoughts about this, that is OK...